From 183396e51a6771d5d2b43f22b0d2a9a91785e533 Mon Sep 17 00:00:00 2001
From: Wouter Deconinck <wdconinc@gmail.com>
Date: Thu, 22 Jun 2023 11:29:03 -0500
Subject: [PATCH] fix: append existing PYTHONPATH to CMAKE_PYTHON_PATH

This addresses #3500. It approaches it slightly differently than the code included there, in order to behave correctly when `PYTHONPATH` is not set and to remain within the CMake version 3.6 minimum (prevents `JOIN`).
---
 src/mlpack/bindings/python/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/mlpack/bindings/python/CMakeLists.txt b/src/mlpack/bindings/python/CMakeLists.txt
index e4b34744748..971cf727d88 100644
--- a/src/mlpack/bindings/python/CMakeLists.txt
+++ b/src/mlpack/bindings/python/CMakeLists.txt
@@ -203,6 +203,9 @@ execute_process(COMMAND ${PYTHON_EXECUTABLE}
     "${PYTHON_INSTALL_PREFIX}"
     OUTPUT_VARIABLE CMAKE_PYTHON_PATH)
 string(STRIP "${CMAKE_PYTHON_PATH}" CMAKE_PYTHON_PATH)
+if (DEFINED ENV{PYTHONPATH})
+    string(APPEND CMAKE_PYTHON_PATH : $ENV{PYTHONPATH})
+endif ()
 install(CODE "set(ENV{PYTHONPATH} ${CMAKE_PYTHON_PATH})")
 install(CODE "set(PYTHON_EXECUTABLE \"${PYTHON_EXECUTABLE}\")")
 install(CODE "set(CMAKE_BINARY_DIR \"${CMAKE_BINARY_DIR}\")")
